### PR TITLE
Make right modal expandable

### LINKED ISSUE #17: Right modal should be expandable
the form size is insufficient and requires too much horizontal scrolling to view data in some cases where large data(table) is shown.
e.g. comment replies, specs etc

### PR DESCRIPTION
**Pull request by piyush-smpx**
fixes #17

### CODE REVIEW

**Comment by piyush-smpx** [comment]
ezgif com-video-to-gif

File: src/RightModal.vue [hunk 1]
```diff
@@ -2,11 +2,12 @@
 	<div v-show="currentValue" class="ela-right-modal-wrapper">
 		<div
 			:style="{
-				maxWidth: `${800 * (1 - (index * 0.016))}px`,
+				maxWidth: `${originalWidth}px`,
 				width: `${90 * (1 - (index * 0.016))}%`,
 				zIndex: `${(index + 1) * 99}`,
 			}"
 			class="ela-right-modal">
+			<div id="drag" @mousedown="dragStart"></div>
```
[the remark this answers is not in the document]
Comment (rohit-smpx) [reply src/RightModal.vue]: Also, will id conflict when multipe modals are open?

File: src/RightModal.vue [hunk 2]
```diff
@@ -31,6 +33,21 @@ export default {
...
+		dragging: false,
+		originalWidth: null,
+		startWidth: 0,
+		pageX: 0,
+		modalEl: '',
+	},
+
+	created() {
+		this.originalWidth = 800 * (1 - (this.index * 0.016));
+		document.addEventListener('mousemove', this.drag);
```
Comment (hit12369) [inline src/RightModal.vue]: These event listeners need to be removed to when component is destroyed.

File: src/RightModal.vue [hunk 3]
```diff
@@ -57,7 +74,34 @@ export default {
...
+
+			const dx = e.pageX - this.pageX;
+			if (this.startWidth - dx < this.originalWidth) return;
+
+			this.modalEl.style.maxWidth = (this.startWidth - dx) + 'px';
+		},
+
+		dragStop(e) {
+			this.dragging = false;
+			document.querySelector('.ela-right-modal').style.userSelect = 'auto';
```
Comment (hit12369) [inline src/RightModal.vue]: Use `ref` instead of `querySelector`

File: src/RightModal.vue [hunk 4]
```diff
@@ -57,7 +74,34 @@ export default {
 
 			this.currentValue = false;
 		},
+
+		dragStart(e) {
+			this.dragging = true;
+			this.modalEl = e.target.parentElement;
+			this.startWidth = parseInt(e.target.parentElement.style.maxWidth);
+			this.pageX = e.pageX;
+			document.querySelector('.ela-right-modal').style.userSelect = 'none';
```
Comment (hit12369) [inline src/RightModal.vue]: Use `ref`

File: src/RightModal.vue [hunk 5]
```diff
@@ -57,7 +74,34 @@ export default {
 
 			this.currentValue = false;
 		},
+
+		dragStart(e) {
+			this.dragging = true;
+			this.modalEl = e.target.parentElement;
```
Comment (hit12369) [inline src/RightModal.vue]: Use `ref`

File: src/RightModal.vue [hunk 6]
```diff
@@ -57,7 +74,34 @@ export default {
...
+			document.querySelector('.ela-right-modal').style.userSelect = 'none';
+		},
+
+		drag(e) {
+			if (!this.dragging) return;
+
+			const dx = e.pageX - this.pageX;
+			if (this.startWidth - dx < this.originalWidth) return;
+
+			this.modalEl.style.maxWidth = (this.startWidth - dx) + 'px';
```
Comment (hit12369) [inline src/RightModal.vue]: Why not change `originalWidth` itself?

**Pull request merged**